### PR TITLE
Refactor/react components

### PR DESCRIPTION
**Pull request by msakr21**
This is an initial round of refactoring.

I would like some input on how to organize the files in the components folder so it's intuitive and a bit more developer friendly, @bkeener7 .

### CODE REVIEW

**Comment by msakr21** [comment]
Please don't merge until you've taken a look at the components folder and we've had a chance to discuss how to organize it!

**Review by bkeener7 — APPROVED** [review]
We went through your changes together. Everything looks good on my end. I added some related commits, going to reverse Uno card the review back to you before merging.

File: spec/features/admin/enrollments/edit_spec.rb [hunk 1]
```diff
@@ -5,40 +5,60 @@
     create(:enrollment, location: 'Eloise May')
   end
 
-  before do
-    visit edit_admin_enrollment_path(eloise_may)
+  let(:date_field_css) do
```
Comment (msakr21) [inline spec/features/admin/enrollments/edit_spec.rb:8]: Absolutely love this!

File: spec/features/admin/enrollments/edit_spec.rb [hunk 2]
```diff
@@ -5,40 +5,60 @@
...
 
-  it "displays a form pre-filled with the enrollment's data" do
-    expect(page).to have_content('Please fill the form below:')
-    expect(page).to have_selector(:css, 'form')
-    expect(page).to have_field('student_limit', with: '30')
-    expect(page).to have_select('location', selected: 'Eloise May')
-    expect(find('input[type="text"][class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"][placeholder="MM/DD/YYYY"]').value).to eq("#{eloise_may.schedule.in_time_zone('Mountain Time (US & Canada)').strftime('%m/%d/%Y')}")
-    expect(find('input[type="text"][class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"][placeholder="hh:mm aa"]').value).to eq("#{eloise_may.schedule.in_time_zone('Mountain Time (US & Canada)').strftime('%I:%M %p')}")
-    expect(page).to have_button('Submit')
+  let(:time_field_css) do
```
Comment (msakr21) [inline spec/features/admin/enrollments/edit_spec.rb:12]: Same as above but I love it so much that here you go again:

File: spec/features/admin/enrollments/index_spec.rb [hunk 3]
```diff
@@ -10,124 +10,140 @@
       visit admin_enrollments_path
     end
 
-    it 'I see a list of enrollments' do
+    it 'displays a list of enrollments' do
       expect(page).to have_content(eloise_may.location)
-      expect(page).to have_content(eloise_may.schedule.in_time_zone('Mountain Time (US & Canada)').strftime('%m/%d/%Y'))
-      expect(page).to have_content(eloise_may.schedule.in_time_zone('Mountain Time (US & Canada)').strftime('%I:%M %p'))
+      expect(page).to have_content(eloise_may.formatted_date)
```
Comment (msakr21) [inline spec/features/admin/enrollments/index_spec.rb:15]: Great change!

**Pull request merged**